charmhelpers/contrib/openstack/context.py

```diff
@@ -2129,29 +2129,17 @@ def __call__(self):
         ctxt = {}
         ctxt['use_memcache'] = enable_memcache(package=self.package)
```

Comment:
if I'm reading this right, this change still installs memcached locally on the unit, and then tries to use the remote cluster?

Also, what happens in the non-HA setup where a remote memcached is unnecessary? Does this change allow the charm to use a local memcached when there is only a single unit but be added to a remote memcached for better HA?

Reply:
Yes, that's correct. To be honest I am not quite sure how to address this, since I found that the Designate charm uses two `memcached` instances, one that is using the `memcached` relation (for `[coordination]`) and one that isn't (for `[keystone_authotoken]`). I have to go through this code a bit more carefully.

I am not sure I understand your second comment. If the charm requires the `memcache` relation then the operator needs to deploy a `memcached` unit. So in even in a non-HA setup we end up with a remote `memcached` instance.

Thanks!

Reply:
Could we perhaps make this conditional on whether the relation actually exists and if not just revert back to the local memcache method?